core/lib/libmbedtls/src/tee_lmd_provider.c

```diff
@@ -767,15 +816,104 @@ void crypto_aes_gcm_final(void *ctx __unused)
 /******************************************************************************
  * Pseudo Random Number Generator
  ******************************************************************************/
-TEE_Result crypto_rng_read(void *buf __unused, size_t blen __unused)
+#if defined(CFG_MBEDTLS_CTR_PRNG)
+static TEE_Result ctr_drbg_read(void *buf, size_t blen)
 {
-	return TEE_ERROR_NOT_IMPLEMENTED;
+	TEE_Result res = TEE_SUCCESS;
+	int err;
+	mbedtls_ctr_drbg_context ctr_drbg;
+
+	mbedtls_ctr_drbg_init(&ctr_drbg);
+
+	err = mbedtls_ctr_drbg_seed(&ctr_drbg, mbd_rand, NULL, NULL, 0);
+	if (err != 0) {
+		EMSG("failed\n ! mbedtls_ctr_drbg_seed returned 0x%x\n", -err);
```

Comment:
Please remove all `failed\n ! ` prefixes in the error strings, the `EMSG()` already tags it as an error in the log.
Also the `\n` at the end of the error messages are not needed, `EMSG()` and friends add those automatically.